gamemodes/terrortown/gamemode/server/sv_entity.lua

```diff
@@ -91,19 +88,16 @@ end
 function entmeta:IsUsableEntity(requiredCaps)
     requiredCaps = requiredCaps or 0
 
-    local caps = self:ObjectCaps()
-
     -- special case: TTT specific lua based use interactions
-    -- when were looking for specifically the lua use, return false it not set
-    if
-        bit.band(FCAP_USE_LUA, requiredCaps) > 0
-        and not (self.CanUseKey or self.player_ragdoll or self:IsWeapon())
-    then
-        return false
-    elseif requiredCaps == 0 and (self.CanUseKey or self.player_ragdoll or self:IsWeapon()) then
+    -- when we're looking for specifically the lua use
+    if self:IsWeapon() or self.player_ragdoll then
+        return true
+    elseif self:IsScripted() and self.CanUseKey then
```

Comment:
```suggestion
    if self:IsWeapon() or self.player_ragdoll or (self:IsScripted() and self.CanUseKey) then
```

Reply:
Not sure anymore if the brackets are needed.

Reply:
I split this into two because I thought it is more readable

Reply:
i second zen's approach. 2 ifs are more confusing IMO

also the parenthesis should not be needed